pkg/activator/test/roundtripper.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/queue"
+)
+
+// FakeResponse is a response given by the FakeRoundTripper
+type FakeResponse struct {
+	Err  error
+	Code int
+	Body string
+}
+
+// FakeRoundTripper is a roundtripper emulator useful in testing
+type FakeRoundTripper struct {
+	T                *testing.T
+	ExpectHost       string
+	LockerCh         chan struct{}
+	ProbeResponses   []FakeResponse
+	RequestResponses []FakeResponse
```

Comment:
I don't see any usage of RequestResponses beyond a single one.
perhaps we only keep probe responses as a slice?